### PR TITLE
New GitHub Actions workflow for backing up Postgres DB

### PR DESCRIPTION
**Pull request by lpzhang-code**
Please don't merge this just yet, some comments and questions attached.

### CODE REVIEW

File: infra/backup.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+HOST='13.55.250.149'
+TIME=$(date "+%s")
+DATABASE_NAME=clerk
+S3_BUCKET=s3://anika-database-backups
+BACKUP_FILE="postgres_${DATABASE_NAME}_${TIME}.sql.gz"
+S3_PATH="$S3_BUCKET/$BACKUP_FILE"
+
+if [[ -z "$CLERK_PRIVATE_SSH_KEY" ]]
+then
+    echo -e "\n>>> Error: Clerk private key not found in CLERK_PRIVATE_SSH_KEY."
```
Comment (lpzhang-code) [inline infra/backup.sh:16]: What is the `-e` flag for? Does this mean that if the echo command fails we should exit the script?

Comment (MattSegal) [reply infra/backup.sh:16]: `-e` means print `\n` as a new line

```enable interpretation of backslash escapes```

Comment (MattSegal) [reply infra/backup.sh:16]: distinct from `set -e` which causes the script to halt on errors

File: infra/backup.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+echo -e "\n>>> Setting up private key."
+echo -e "$CLERK_PRIVATE_SSH_KEY" > private.key
+chmod 600 private.key
+
+echo -e "\n>>> SSHing into Clerk EC2 instance at $HOST."
+ssh -o StrictHostKeyChecking=no -i private.key root@$HOST /bin/bash << EOF
+    set -e
+    cd /srv/backups
+    . env/bin/activate
+    touch clerk.log
```
Comment (MattSegal) [inline infra/backup.sh:31]: both the virtualenv and the logfile aren't required anymore

- virtualenv: `pg_dump` and `aws` are both already installed on the clerk box so there's no need for this
- logfile: github actions will store action logs for us (for a while at least)

please remove the use of virtualenv and logfiles

File: .github/workflows/backup.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: Backup Postgres
+on: [workflow_dispatch]
```
Comment (MattSegal) [inline .github/workflows/backup.yml:2]: we can move this to a cron later - but we do want this to run every night

File: .github/workflows/backup.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+jobs:
+  backup:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run backup script
+      run: ./infra/backup.sh
+      shell: bash
+      env:
+        CLERK_PRIVATE_SSH_KEY: ${{ secrets.CLERK_PRIVATE_SSH_KEY }}
```
Comment (MattSegal) [inline .github/workflows/backup.yml:12]: newline at the ends of files pls

File: infra/backup.sh (hunk 2 above)
Comment (lpzhang-code) [reply infra/backup.sh:31]: @MattSegal 
- we currently SSH into our Clerk instance and then write detailed information to `clerk.log`
- how does GA store logs for us? I assume that any info from operations we perform whilst we have an SSH connection to Clerk instance will not be logged by GA. 
- I assume that GA logs info from commands such as `echo -e "Have done such and such"` - in that case we might need to insert additional echo statements?

File: .github/workflows/backup.yml (hunk 3 above)
Comment (MattSegal) [reply .github/workflows/backup.yml:2]: to clarify I meant a github actions cron schedule
eg. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

File: infra/backup.sh (hunk 2 above)
Comment (MattSegal) [reply infra/backup.sh:31]: hey Peter all stdout, including that from a SSH session should be captured by githhub actions and be accessible via the github UI

**Pull request merged**